### ISSUE TITLE
Navigation Screen: Always show appender

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#31276]
## What problem does this address?

The default menu on the navigation screen doesn't show the appender:

<img width="655" alt="Screenshot 2021-04-28 at 12 28 05" src="https://user-images.githubusercontent.com/1204802/116390012-e9811180-a81d-11eb-9441-dbb4670095e4.png">

Not until the menu is selected:

<img width="687" alt="Screenshot 2021-04-28 at 12 28 35" src="https://user-images.githubusercontent.com/1204802/116389995-e554f400-a81d-11eb-8dd2-90295b9cedf9.png">

This is the default behavior for inner-blocks, and the the navigation, even on this screen, is technically a block.

## What is your proposed solution?

Extracting from https://github.com/WordPress/gutenberg/pull/31228#issuecomment-827964020, maybe that plus should show up by default, even without the navigation block being selected first.

**Comment by jasmussen** [comment]
CC: @javierarce in case you have any thoughts on a better approach!

**Comment by javierarce** [comment]
I think there's an indirect and less "invasive" way to solve this issue. If we fixed the hover state of the menu highlighting its border as I mentioned [here](https://github.com/WordPress/gutenberg/issues/30659), users would naturally understand that the menu is clickable.

That said, I think your solution could work, and it's probably more accessible (although from a purely aesthetic perspective and to maintain the default behaviour, I wouldn't show an item that's not part of the menu).

**Comment by paaljoachim** [comment]
This is what I see with Gutenberg 10.5.3 activated.

![Screen Shot 2021-05-04 at 18 54 37](https://user-images.githubusercontent.com/5323259/117040496-3ecf8e00-ad0a-11eb-8bde-bfea883de3a9.png)

EDIT:
I left the screen and came back again. This time no block inserter was seen by default.

![Screen Shot 2021-05-04 at 18 58 54](https://user-images.githubusercontent.com/5323259/117041019-cae1b580-ad0a-11eb-85c8-8b3eac6cc16b.png)

Having the + icon button appender shows that the area can be interacted with. It is like a call for action to the user.
I do believe it will be helpful to have it on by default.

**Comment by spacedmonkey** [comment]
Maybe the best solution to this problem, is to add the block inserter to navigation screen? This would keep the navigation screen inline other block editors. 

#### Post editor
<img width="411" alt="Screenshot 2021-08-30 at 23 24 31" src="https://user-images.githubusercontent.com/237508/131413861-b0311785-de6b-4678-a9e0-7c91e732e569.png">

#### Widget editor 
<img width="418" alt="Screenshot 2021-08-30 at 23 24 15" src="https://user-images.githubusercontent.com/237508/131413887-68965dd9-7d81-4ebd-981d-057df2c3df98.png">

#### Navigation editor
<img width="269" alt="Screenshot 2021-08-30 at 23 25 04" src="https://user-images.githubusercontent.com/237508/131413927-4d8029cb-8c24-43ca-b976-1242dd9866b3.png">

**Comment by talldan** [comment]
This has become more important since #34568 was merged.

Before that the navigation editor showed the FSE-style placeholder preview for an empty unselected navigation block:
<img width="444" alt="Screenshot 2021-09-06 at 3 49 11 pm" src="https://user-images.githubusercontent.com/677833/132179939-e133285a-449d-400c-9a01-e390c1cc233f.png">

That was undesired, but what happens now is there's just an empty block:
<img width="600" alt="Screenshot 2021-09-09 at 1 03 48 pm" src="https://user-images.githubusercontent.com/677833/132625830-8929fd6c-839a-4154-9f27-ddd4c240ce86.png">

Fixing this does mean fighting with a lot of the block appender visibility logic. It has both code and styles for hiding the appender, most of which need to be revisited.
